### PR TITLE
Upgrade to babel 7, and include Prettier optimization

### LINKED ISSUE #1312: Pass Babel AST to Prettier rather than re-parsing
See _TODO_ comments in `js/repl/src/compile`.

It would be more efficient to pass Prettier the AST Babel has already created. However we can't do this until we upgrade to Babel 7 (since that's what Prettier uses).

It also seems suboptimal that Prettier only exposes this functionality behind a `__debug` property (`prettier.__debug.formatAST`). Maybe we could reach out to the Prettier team about a more formal API for this?

### PR DESCRIPTION
**Pull request by azz**
Closes #1312

At the moment the build fails with 

```
TypeError: Property declaration of ExportNamedDeclaration expected node to be of a type ["Declaration"] but instead got "ExpressionStatement"
```

Maybe related to https://github.com/facebook/regenerator/pull/319?

<details>
<summary>Build Error</summary>

```
ERROR in ./js/repl/loadBabel.js
Module build failed: TypeError: Property declaration of ExportNamedDeclaration expected node to be of a type ["Declaration"] but instead got "ExpressionStatement"
    at Object.validate (/Users/azz/code/babel-website/node_modules/@babel/types/lib/definitions/utils.js:88:13)
    at Object.validate (/Users/azz/code/babel-website/node_modules/@babel/types/lib/validators/validate.js:15:9)
    at NodePath._replaceWith (/Users/azz/code/babel-website/node_modules/@babel/traverse/lib/path/replacement.js:169:7)
    at NodePath.replaceWith (/Users/azz/code/babel-website/node_modules/@babel/traverse/lib/path/replacement.js:153:8)
    at Object.replaceWithOrRemove (/Users/azz/code/babel-website/node_modules/regenerator-transform/lib/util.js:49:10)
    at PluginPass.<anonymous> (/Users/azz/code/babel-website/node_modules/regenerator-transform/lib/visit.js:156:16)
    at PluginPass.<anonymous> (/Users/azz/code/babel-website/node_modules/regenerator-transform/lib/util.js:27:17)
    at newFn (/Users/azz/code/babel-website/node_modules/@babel/traverse/lib/visitors.js:223:21)
    at NodePath._call (/Users/azz/code/babel-website/node_modules/@babel/traverse/lib/path/context.js:64:19)
    at NodePath.call (/Users/azz/code/babel-website/node_modules/@babel/traverse/lib/path/context.js:38:17)
 @ ./js/repl/Repl.js 26:40-62
 @ ./js/repl/index.js
 @ multi (webpack)-dev-server/client?http://localhost:34512 webpack/hot/dev-server ./js/repl/index.js
```
</details>

Any ideas?

### CODE REVIEW

**Comment by existentialism** [comment]
@azz I think that error should be fixed by bumping to beta 35

File: js/repl/loadBabel.js [hunk 1]
```diff
@@ -3,7 +3,7 @@ import { loadBuildArtifacts, loadLatestBuildNumberForBranch } from "./CircleCI";
 import { BabelState } from "./types";
 import WorkerApi from "./WorkerApi";
 
-const DEFAULT_BABEL_VERSION = "6";
+const DEFAULT_BABEL_VERSION = "7";
```
Comment (azz) [inline js/repl/loadBabel.js]: This won't work until v7 of `@babel/standalone` is released

File: package.json [hunk 2]
```diff
@@ -58,7 +59,7 @@
     "webpack-dev-server": "^2.7.1"
   },
   "dependencies": {
-    "babel-plugin-emotion": "^8.0.2-11",
+    "babel-plugin-emotion": "^8.0.12",
```
Comment (azz) [inline package.json]: This is the latest tagged version and still uses babel v6.

**Comment by azz** [comment]
Thanks @existentialism, that got sorted it.

Do we want to wait until babel 7 lands before landing this change?

**Comment by hzoo** [comment]
@azz not really since we are using v7 in Babel v7, don't see why we should wait, if it helps test it more

**Comment by azz** [comment]
Confused as why it loads beta.34 instead of beta.35 (DEFAULT_BABEL_VERSION).

Each time I reload it loads the next oldest version...

**Comment by existentialism** [comment]
@azz re: version number https://github.com/babel/babel/issues/6174

**Comment by azz** [comment]
Ok, I'm not crazy. Thanks.
<details>
<summary></summary>
Well, not about this at least.
</details>

**Pull request closed without merging**